silence/silence.go

```diff
@@ -217,18 +215,19 @@ type Limits struct {
 type MaintenanceFunc func() (int64, error)
 
 type metrics struct {
-	gcDuration              prometheus.Summary
-	snapshotDuration        prometheus.Summary
-	snapshotSize            prometheus.Gauge
-	queriesTotal            prometheus.Counter
-	queryErrorsTotal        prometheus.Counter
-	queryDuration           prometheus.Histogram
-	silencesActive          prometheus.GaugeFunc
-	silencesPending         prometheus.GaugeFunc
-	silencesExpired         prometheus.GaugeFunc
-	propagatedMessagesTotal prometheus.Counter
-	maintenanceTotal        prometheus.Counter
-	maintenanceErrorsTotal  prometheus.Counter
+	gcDuration                prometheus.Summary
+	snapshotDuration          prometheus.Summary
+	snapshotSize              prometheus.Gauge
+	queriesTotal              prometheus.Counter
+	queryErrorsTotal          prometheus.Counter
+	queryDuration             prometheus.Histogram
+	silencesActive            prometheus.GaugeFunc
+	silencesPending           prometheus.GaugeFunc
+	silencesExpired           prometheus.GaugeFunc
+	propagatedMessagesTotal   prometheus.Counter
+	maintenanceTotal          prometheus.Counter
+	maintenanceErrorsTotal    prometheus.Counter
+	matcherCompileErrorsTotal *prometheus.CounterVec
```

Comment:
Minor suggestion.

In the struct, declare two `prometheus.Counter()` values.

Then in the `newMetrics()` declare the `NewCounterVec()` and assign the label values.